internal/notifier/client.go

```diff
@@ -32,29 +32,90 @@ import (
 	"github.com/hashicorp/go-retryablehttp"
 )
 
-type requestOptFunc func(*retryablehttp.Request)
+type postOption struct {
+	proxy             string
+	certPool          *x509.CertPool
+	requestModifiers  []requestModifier
+	responseValidator func(*http.Response) bool
+}
+type requestModifier func(*retryablehttp.Request)
```

Comment:
Either define types for both the request modifier and the response validator, or don't define those types. I'm fine with not defining either